src/lib/utils/index.ts

```diff
@@ -1,11 +1,19 @@
 export * from './domains';
 
 interface DomainMetadataParams {
+	previewImage?: Buffer;
 	image: Buffer;
 	name: string;
 	story: string;
 }
 
+interface MetadataObject {
```

Comment:
It was smart to create a helper object here, however an object that fulfills this purpose already exists. If you look in `types.ts` https://github.com/zer0-os/zNS-dapp/blob/3e4f579e814e843981a7220b0a151fc34e04a899/src/lib/types.ts#L20

You will see a `Metadata` interface that gets declared. This is the interface which **all** metadata objects should follow.

Please use that interface instead of creating your own.

Additionally; with this object you have created, you have introduced a bug into the dApp. You have renamed `image` to `imageUri`.
The `Metadata` interface I mentioned is the interface which all metadata objects get casted as when they are **read** from IPFS. The code you are writing is the code that **writes** to IPFS. Read/Write operations to the same data should share the same interface (as mentioned above) but when they don't, bugs like this can happen where you write one set of data, but then try to read a different set of data. So if this code was used, all images would be broken on read.